Lewd viro

ktccd opened the pull request:
:cl: ktccd
add: Added new citadel-only symptoms to viruses.
/:cl:

[why]: # 
Because we got these citadel-only things which could be used by virology. Mostly came to think about it when I figured if there was some way to get airborne aphrodisiac, when the word "airborne" made me think of virology.
# Ty to Snugglesnek for giving the idea of a virus that messes with genitalia. Sadly, the way we handle genitalia is very snow-flakey and not generic for all genitals, so yeah...

ktccd committed:
Adds citadel virus symptoms

Adds a few symptoms that use citadel code stuff.
Currently disabled breast-related code, since there's a big rework PR in
progress that would seriously mess with it.

ktccd committed:
Smaller changes

Reworked the out-commented breastcode to work better and messed with the
speed of the genital symptoms.

TalkingCactus commented on code/citadel/cit_virology.dm:
```diff
@@ -0,0 +1,244 @@
+/datum/symptom/macro_genitals/proc/grow_genitalia(mob/living/carbon/human/M)
+	if(M.has_penis())
+		var/obj/item/organ/genital/penis/O = M.getorganslot("penis")
+		if(O.length < COCK_SIZE_MAX) //Couldn't find a macro or anything for the max size of a dong, so hardcoded, sadly.
```
Useless comment, defines save on memory and are easy to maintain for constants that are referenced a lot.

ktccd replied:
yeah, I think I forgot to remove the comment after finding the constant. When I wrote the comment, I had just put "20" there XD.

ktccd replied:
✅

TalkingCactus commented on code/citadel/cit_virology.dm:
```diff
@@ -0,0 +1,244 @@
+	if(M.has_penis())
+		var/obj/item/organ/genital/penis/O = M.getorganslot("penis")
+		if(O.length < COCK_SIZE_MAX) //Couldn't find a macro or anything for the max size of a dong, so hardcoded, sadly.
+			O.length = O.length + 1
```
`O.length = Clamp(O.length++, COCK_SIZE_MIN, COCK_SIZE_MAX)`
or
```
O.length++
if(O.length > COCK_SIZE_MAX) O.length = COCK_SIZE_MAX
```

ktccd replied:
so clamp forces it to stay between the boundaries?

TalkingCactus replied:
Yes

ktccd replied:
✅

TalkingCactus commented on code/citadel/cit_virology.dm:
```diff
@@ -0,0 +1,244 @@
+/datum/symptom/macro_genitals/Activate(datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		//I think genitalia code is implemented on human level only, so let's not mess up
```
`if(ishuman(A.affected_mob))`

ktccd replied:
oh, yeah. might be weird to recast non-humans to humans. heh...

ktccd replied:
✅

TalkingCactus commented on code/citadel/cit_virology.dm:
```diff
@@ -0,0 +1,244 @@
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		//I think genitalia code is implemented on human level only, so let's not mess up
+		var/mob/living/carbon/human/M = A.affected_mob
+		if(M)
```
Move this below the `..()` and change it to 
```
if(!A.affected_mob) return
```
Though I don't think this proc would even call if there wasn't an affected mob.

ktccd replied:
I think my idea... was something like, M would be null if A wasn't human. But I was just bad at how recasting types worked. I think maybe the result of recasting A as human, if it wasn't human, would probably not be null after all.

TalkingCactus replied:
Monkeys can carry viruses as well. Xenos might be able to as well, haven't tested that in a while.

ktccd replied:
so there's no point in doing if(M) then.
I guess it makes sense, none of the other viruses seem to do that.

TalkingCactus replied:
Mhmm

ktccd replied:
✅

TalkingCactus commented on code/citadel/cit_virology.dm:
```diff
@@ -0,0 +1,244 @@
+	if(M.has_penis())
+		var/obj/item/organ/genital/penis/O = M.getorganslot("penis")
+		if(O.length > COCK_SIZE_MIN) //Couldn't find a macro or anything for the max size of a dong, so hardcoded, sadly.
+			O.length = O.length - 1
```
See Extendong comment; Same thing but inverse.

ktccd replied:
✅

ktccd committed:
Merge remote-tracking branch 'refs/remotes/Citadel-Station-13/master' into Lewd_Viro

ktccd committed:
Implementing feedback fixes

Also moved the procs for genital size-change to the genitals themselves.
Breasts code refactored to behave more like other genitals do with
sizes, i.e using numbers for size and not strings.
Defines are also used for this, so the max breast size define has been
updated to reflect the current size restrictions.

ktccd committed:
Resolving conflicts

This is the master-versions of the files. I will need to make a new PR
with the new versions of the files, because github freaking sucks are
resolving non-text conflicts...

TalkingCactus commented on code/citadel/organs/genitals.dm:
```diff
@@ -32,6 +32,12 @@
+
+/obj/item/organ/genital/proc/shrink_size()
+
+
```
These both should be combined into `AdjustSize(amt)` so it can accept an argument for changing the size. Look at the arousal setting procs for clues for achieving this. This proc would do `organ.size = (organ.size + amt)` and clamp it within the bounds set through the defines in `cit_defines.dm`. Doing this allows you to do increase and decreases in the same proc.

After that, add another proc called `SetSize(size)` which can be used to force a size change to a specific point.

ktccd committed:
Fixing feedback and more stuff

Reintroducing conflicts, 'cause github hates me.

ktccd committed:
Conflict resolving breasts_onmob.dmi

Just sending in the unchanged version.

ktccd committed:
Merge branch 'master' into Lewd_Viro

ktccd committed:
Merge branch 'master' into Lewd_Viro

Poojawa committed:
Merge branch 'master' into Lewd_Viro
